src/components/tooltip/index.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { FC, ReactNode, useState, useRef } from "react"
+
+import PopUp from "./popUp"
+
+export enum TooltipPosition {
+  top = "top",
+  topLeft = "topLeft",
+  topRight = "topRight",
+  bottom = "bottom",
+  bottomLeft = "bottomLeft",
+  bottomRight = "bottomRight",
+  left = "left",
+  right = "right",
+}
+
+const spacingPosition = {
+  top: "top",
+  topLeft: "top",
+  topRight: "top",
+  bottom: "bottom",
+  bottomLeft: "bottom",
+  bottomRight: "bottom",
+  left: "left",
+  right: "right",
+}
+
+const oppositePosition = {
```

Comment:
To me, it feels very counter-intuitive, probably from working with other tooltip libraries to switch directions 180deg when not enough space is available id rather expect them to top and then bottom as there, in theory, must be space in like 99isch per cent of the cases whereas left or right will be to small way more often if the page is to small

Reply:
My idea was that this is supposed to solve the case when the content is scrolled too far to accommodate for tooltip.
I want to limit the logic of checking the available space to this case and make the consumer handle responsive cases by adjusting the position accordingly. Does that make sense?